clippy_tests/examples/range.rs

```diff
@@ -29,7 +21,4 @@ fn main() {
     let v2 = vec![4,5];
     let _x = v1.iter().zip(0..v1.len());
     let _y = v1.iter().zip(0..v2.len()); // No error
-
-    // check const eval
-    let _ = v1.iter().step_by(2/3);
```

Comment:
I want to do so, but directly passing -1 or 0 to `step_by` causes a compile error immediately, which will suppress the clippy warnings. I don't think that's what we expected.